Userland/Libraries/LibWeb/CSS/SelectorEngine.cpp

```diff
@@ -351,14 +351,14 @@ static inline bool matches(CSS::Selector::SimpleSelector const& component, DOM::
     case CSS::Selector::SimpleSelector::Type::Universal:
         return true;
     case CSS::Selector::SimpleSelector::Type::Id:
-        return component.name() == element.attribute(HTML::AttributeNames::id);
+        return component.name().bytes_as_string_view() == element.attribute(HTML::AttributeNames::id);
     case CSS::Selector::SimpleSelector::Type::Class:
-        return element.has_class(component.name());
+        return element.has_class(component.name().bytes_as_string_view());
     case CSS::Selector::SimpleSelector::Type::TagName:
         // See https://html.spec.whatwg.org/multipage/semantics-other.html#case-sensitivity-of-selectors
         if (element.document().document_type() == DOM::Document::Type::HTML)
-            return component.lowercase_name() == element.local_name();
-        return component.name().equals_ignoring_case(element.local_name());
+            return component.lowercase_name().bytes_as_string_view() == element.local_name();
+        return component.name().bytes_as_string_view().equals_ignoring_case(element.local_name());
```

Comment:
I think this in particular should use String's APIs, unless these are meant to be ASCII comparisons (in which case they should use the ASCII comparator you recently added :upside_down_face: )

Reply:
It's ASCII again! :partying_face: